standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetastoreDirectSqlUtils.java

```diff
@@ -152,9 +154,18 @@ static <T> int loopJoinOrderedResult(PersistenceManager pm, TreeMap<Long, T> tre
           if (nestedId < id) {
             throw new MetaException("Found entries for unknown ID " + nestedId);
           }
-          if (nestedId > id)
+          if (nestedId > id) {
+            if (!tree.containsKey(nestedId)) {
+              Throwable throwable = (new Throwable()).fillInStackTrace();
```

Comment:
why do we need to create Throwable?

Reply:
it helps to know what the `many` is (one-to-many mapping), for example if the `value` is Partition, then the `many` can be parameters or partition values.

Reply:
that's fine to log the `value.getClass().getSimpleName()`, however, i don't see benefit of `throwable` here

Reply:
not enough, for example, if the value is a `Partition`, then what are missing for this `nestedId` Partition, it might be the values stored in `PARTITION_KEY_VALS` or the parameters in `PARTITION_PARAMS`, if we shows the message who makes the call, it will direct us to the table quickly to check if this table is corrupted or not.